treetools/treeinfo/treeinfo.cpp

```diff
@@ -271,20 +271,37 @@ int main(int argc, char *argv[])
     new_attributes.push_back("pos_in_branch");
     new_attributes.push_back("segment_length");
   }
-  const int volume_id = num_attributes + 0;
-  const int diameter_id = num_attributes + 1;
-  const int length_id = num_attributes + 2;
-  const int strength_id = num_attributes + 3;
-  const int min_strength_id = num_attributes + 4;
-  const int dominance_id = num_attributes + 5;
-  const int angle_id = num_attributes + 6;
-  const int children_id = num_attributes + 7;
-  // optional with branch_data
-  const int branch_id = num_tree_attributes + 8;
-  const int branch_order_id = num_tree_attributes + 9;
-  const int extension_id = num_tree_attributes + 10;
-  const int pos_in_branch_id = num_tree_attributes + 11;
-  const int segment_length_id = num_tree_attributes + 12;
+
+  // Indices for new attributes
+  int index = num_attributes; // Start indexing from the number of existing attributes
+  const int volume_id = index++;
+  const int diameter_id = index++;
+  const int length_id = index++;
+  const int strength_id = index++;
+  const int min_strength_id = index++;
+  const int dominance_id = index++;
+  const int angle_id = index++;
+  const int children_id = index++;
+
+  // Declare optional attribute indices
+  int branch_id = -1;
```

Comment:
index++ is nice and having those default to -1 when branch_data is not set is nice for safety, but it doesn't fix any bugs as far as I can see, because branch_id, branch_order_id etc are only every used within an if(branch_data.isSet()) block.